### PR TITLE
THORN-2067: Switch to smallrye-rest-client implementation.

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by mkouba** [comment]
Windows failures:
```
Test Result (3 failures / +1)
org.wildfly.swarm.management.console.ManagementConsoleTest.testHomePageShouldHaveCorrectTitle
org.eclipse.microprofile.fault.tolerance.tck.bulkhead.BulkheadAsynchRetryTest.testBulkheadMethodAsynchronousRetry55
org.eclipse.microprofile.fault.tolerance.tck.bulkhead.BulkheadAsynchRetryTest.testBulkheadPassiveRetryMethodAsynchronous55
```
`BulkheadAsynchRetryTest` failures are expected. I'm not sure about `ManagementConsoleTest`.

**Comment by kenfinnigan** [comment]
It can be ignored as well as it always seems to have issues on Windows on occasion

File: fractions/microprofile/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/RestClientFraction.java [hunk 1]
```diff
@@ -15,17 +15,16 @@
  */
 package org.wildfly.swarm.microprofile.restclient;
 
+import static org.wildfly.swarm.spi.api.annotations.DeploymentModule.MetaInfDisposition.IMPORT;
```
Comment (sberyozkin) [inline fractions/microprofile/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/RestClientFraction.java]: Is this still needed ? My basic understanding is that it is only needed if some work is being done in the 'deployment' package

File: fractions/microprofile/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/RestClientFraction.java [hunk 2]
```diff
@@ -15,17 +15,16 @@
...
 import org.wildfly.swarm.spi.api.annotations.DeploymentModule;
 
-import static org.wildfly.swarm.spi.api.annotations.DeploymentModule.MetaInfDisposition.IMPORT;
-
 /**
  * @author Heiko Braun
  */
-@DeploymentModule(name = "org.wildfly.swarm.microprofile.restclient", slot = "deployment", metaInf = IMPORT,
-        export = true, services = Module.ServiceHandling.IMPORT)
+@DeploymentModule(name = "io.smallrye.restclient", metaInf = IMPORT, export = true, services = Module.ServiceHandling.IMPORT)
```
Comment (sberyozkin) [inline fractions/microprofile/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/RestClientFraction.java:27]: Is this still needed or updating the pom dependency and module.xml is enough ?

Comment (sberyozkin) [reply fractions/microprofile/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/RestClientFraction.java:27]: Based on Michal's feedback to my comment to his PR I guess my comment can simply be ignored :-)

**Review by sberyozkin — APPROVED** [review]
LGTM

**Comment by sberyozkin** [comment]
Unrelated Win test failure, going ahead with the merge

**Pull request merged**